src/cargo/ops/cargo_compile/mod.rs

```diff
@@ -542,6 +629,77 @@ where `<compatible-ver>` is the latest version supporting rustc {rustc_version}"
     Ok(bcx)
 }
 
+fn get_specific_check_cfg_for_unit<'a>(
+    gctx: &GlobalContext,
+    rustc: &Rustc,
+    unit: &Unit,
+) -> CargoResult<Option<CheckCfg>> {
+    let lint_rustflags = unit.pkg.manifest().lint_rustflags();
+
+    if lint_rustflags.iter().any(|a| a == "--check-cfg") {
+        Ok(Some(CheckCfg::new(gctx, rustc, Some(lint_rustflags))?))
```

Comment:
I don't feel comfortable having us parse a command line

Reply:
I guess we don't parse it but forward it to rustc but then we are conditionally including unrelated flags in the call to rustc which feels odd and potentially brittle

Reply:
Yeah, it's a hack and I don't really like it either, but we just deduplicated all the logic in https://github.com/rust-lang/cargo/pull/14567, that I didn't wanted to reintroduce one that will potentially stay indefinitely and the unrelated flags are only allow/warn/deny which is fine for `--print=check-cfg`.